packages/x-components/src/components/base-keyboard-navigation.vue

```diff
@@ -32,116 +32,128 @@
    *
    * @public
    */
-  @Component
-  export default class BaseKeyboardNavigation extends Vue {
-    /**
-     * An array of {@link TakeNavigationControl} objects defining when to
-     * take control of the keyboard navigation.
-     */
-    @Prop({
-      default: () => [
-        { xEvent: 'UserPressedArrowKey', moduleName: 'searchBox', direction: 'ArrowDown' }
-      ]
-    })
-    protected navigationHijacker!: TakeNavigationControl[];
-
-    /**
-     * An {@link EventsForDirectionLimit} to emit when the user is already at the furthest element
-     * in a direction and tries to keep going on the same direction.
-     */
-    @Prop({ default: () => ({ ArrowUp: 'UserReachedEmpathizeTop' }) })
-    protected eventsForDirectionLimit!: Partial<EventsForDirectionLimit>;
-
-    /**
-     * The {@link SpatialNavigation} service to use.
-     */
-    protected navigationService!: SpatialNavigation;
-
-    /**
-     * The element to focus.
-     */
-    protected elementToFocus: HTMLElement | undefined;
-
-    mounted(): void {
-      // TODO Replace this with injection
-      this.navigationService = new DirectionalFocusNavigationService(this.$el as HTMLElement);
-    }
-
-    /**
-     * Get the navigation hijacker events.
-     *
-     * @remarks
-     * If the same {@link XEvent} is defined multiple times it is only inserted once.
-     *
-     * @returns The events to hijack the navigation.
-     */
-    protected get navigationHijackerEvents(): XEventsOf<ArrowKey>[] {
-      const eventsSet = this.navigationHijacker.map(({ xEvent }) => xEvent);
-      return Array.from(new Set(eventsSet));
-    }
-
-    /**
-     * Trigger navigation if this component is in control of it.
-     *
-     * @param eventPayload - The {@link @empathyco/x-bus#SubjectPayload.eventPayload}.
-     * @param metadata - The {@link @empathyco/x-bus#SubjectPayload.metadata}.
-     * @public
-     */
-    @XOn(component => (component as BaseKeyboardNavigation).navigationHijackerEvents)
-    triggerNavigation(eventPayload: ArrowKey, metadata: WireMetadata): void {
-      if (this.hasToTakeNavigationControl(eventPayload, metadata)) {
-        this.focusNextNavigableElement(eventPayload);
+  export default defineComponent({
+    name: 'BaseKeyboardNavigation',
+    props: {
+      /**
+       * An array of {@link TakeNavigationControl} objects defining when to
+       * take control of the keyboard navigation.
+       */
+      navigationHijacker: {
+        type: Array as PropType<TakeNavigationControl[]>,
+        default: () => [
+          { xEvent: 'UserPressedArrowKey', moduleName: 'searchBox', direction: 'ArrowDown' }
+        ]
+      },
+      /**
+       * An {@link EventsForDirectionLimit} to emit when the user is already at the furthest element
+       * in a direction and tries to keep going on the same direction.
+       */
+      eventsForDirectionLimit: {
+        type: Object as PropType<EventsForDirectionLimit>,
```

Comment:
I think the type here should be 
```suggestion
        type: Object as PropType<Partial<EventsForDirectionLimit>>,
```